spec/datadog/tracing/contrib/mysql2/patcher_spec.rb

```diff
@@ -103,6 +104,71 @@
             .to eq("Unknown column 'INVALID' in 'field list'")
         end
       end
+
+      context 'when sql comment propagation' do
+        let(:sql_statement) { 'SELECT 1' }
+
+        subject { client.query(sql_statement) }
+
+        shared_examples_for 'propagated with sql comment propagation' do |mode, span_op_name|
```

Comment:
These shared examples are probably also compatible with testing `pg` and other integrations that will be affected by this SqlComment propagation.

We don't need to do this for this PR, but I recommend sharing a few `shared_examples_for` when we add `pg`, for example.